### PR TITLE
spinW integration to Horace.

### LINKED ISSUE #1745: Energy cuts and Horace integration problem (Horace 4 and SpinW interface problem)
Dear Horace Development Team,

I am encountering an issue when performing cuts on a SpinW model using Horace 4. Errors occur during the process (please see the attached figure), though everything works fine with Horace 3. I believe the issue may be related to Horace 4's new cutting algorithm.

To reproduce the bug, you can run Tutorial 10 ("Constant energy cut using Horace") from SpinW. The tutorial can be accessed via the following link: https://spinw.org/tutorials/10tutorial.

Best regards,
Zhaoyang
图片5

### PR DESCRIPTION
**Pull request by abuts**
fixes #1745.

The changes add new (or rather recovers old) spinW-style constructor to `DnD` objects and unit tests for it. It also modifies `disp2sqw` to be able to produce any `spinW` generated plots. 

unlike old interface, disp2sqw requests invoking option `'-all'` to produce valid `dnd` objects compartible with `spinW`

This work should be accompanied by changes in integration tests.

### CODE REVIEW

File: horace_core/sqw/@DnDBase/private/init_.m [hunk 1]
```diff
@@ -128,8 +139,9 @@
         error(['HORACE:', class(obj),':invalid_argument'], ...
             'Unrecognised number or type of the input arguments')
     end
-
 else
     error(['HORACE:', class(obj),':invalid_argument'], ...
         'unknown input for %s constructor',class(obj));
 end
+
+function ob
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@DnDBase/private/init_.m:147]: looks like this hasn't been cleaned up

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]
One incompletely cleaned up line, otherwise ok, modulo Duc approving

File: horace_core/sqw/@DnDBase/private/init_.m (hunk 1 above)
Comment (abuts) [reply horace_core/sqw/@DnDBase/private/init_.m:147]: yes, I have apparently forgot to  committ the clean-up

**Review by cmarooney-stfc — APPROVED** [review]

**Comment by abuts** [comment]
fixed through Re #1757

**Pull request merged**